tokenize_data.py

```diff
@@ -0,0 +1,143 @@
+            return_attention_mask=False)
+
+    entire_dataset = entire_dataset.map(tokenization, batched=True)
+
+    # Drop now unnecessary text_feature column
+    entire_dataset = entire_dataset.remove_columns(column_names=text_feature)
+
+    #This code saves the now tokenized dataset as a .parquet folder, making a folder in the data directory called tokenized if one does not already exist.
+    path = Path(tokenized_data_folder)
+    print(f'Saving tokenized data to {path}')
+    if not path.exists():
+        path.mkdir(parents=True)
+
+    if isinstance(entire_dataset, datasets.arrow_dataset.Dataset):
+        entire_dataset.to_parquet(path  / f'{tokenized_data_name}.parquet')
+    elif isinstance(entire_dataset, datasets.dataset_dict.DatasetDict):
+        for key, value in entire_dataset.items():
+            filename = key + '.parquet'
+            value.to_parquet(path / filename)
+    else:
+        print('Dataset is not of type datasets.arrow_dataset.Dataset or datasets.dataset_dict.DatasetDict')
+
+
+if __name__ == "__main__":
+    # Get arguments
+    parser = ArgumentParser()
+
+    parser.add_argument(
+        "--tokenized_data_name",
+        type=str,
+        required=True,
+        help="Name of tokenized data.")
+
+    parser.add_argument(
+        "--tokenized_data_folder",
+        type=str,
+        required=True,
+        help="Folder to save tokenizered data to.")
+    parser.add_argument(
+        "--tokenizer_folder",
+        type=str,
+        required=True,
+        help="Folder to save tokenizer to.")
+    parser.add_argument(
+        "--dataset_name",
+        type=str,
+        required=True,
+        help="Name of Hugging Face dataset.")
+    parser.add_argument(
+        "--seq_len",
+        type=int,
+        required=True,
+        help="Context window/sequence length.")
+    parser.add_argument(
+        "--dataset_dir",
+        type=str,
+        required=True,
+        help="Relative path from base of repository to directory in which to download the dataset.")
+    parser.add_argument(
+        "--dataset_subset",
+        type=str,
+        default=None,
+        help="Configuration/subset of dataset to use.")
+    parser.add_argument(
+        "--text_feature",
+        type=str,
+        default="text",
+        help="Name of the feature/column of the dataset to use.")
+    parser.add_argument("--splits", type=float, nargs=3,
+        default=[0.7, 0.2, 0.1],
+        help="Space-separated decimal splits of train, validation, and " + \
+            "test datasets. (Ex: '0.7 0.2 0.1')")
+    parser.add_argument(
+        "--rand_seed",
+        type=int,
+        default=None,
+        help="Seed used during dataset shuffling, ignored if None.")
+
+    args = parser.parse_args()
+
+    tokenize_data(args.tokenized_data_name, args.tokenized_data_folder, args.tokenizer_folder, args.dataset_name, args.seq_len, args.dataset_dir, args.dataset_subset, args.text_feature, args.splits, args.rand_seed)
```

Comment:
See code in train_model.py and grid_search.py for a cleaner way to pass in the args. However, this looks like it will work.